File: src/main/java/cn/labzen/logger/reload4j/LabzenReload4jLoggerFactory.java
package cn.labzen.logger.reload4j;

import cn.labzen.logger.kernel.LabzenLogger;
import org.slf4j.ILoggerFactory;
import org.slf4j.Logger;

public class LabzenReload4jLoggerFactory implements ILoggerFactory {

  private final ILoggerFactory principal;

  public LabzenReload4jLoggerFactory(ILoggerFactory principal) {
    this.principal = principal;
  }

  @Override
  public Logger getLogger(String name) {
    Logger original = principal.getLogger(name);
    return new LabzenLogger(original);
  }
}

File: src/main/kotlin/cn/labzen/logger/reload4j/LabzenReload4jLoggerFactory.kt
package cn.labzen.logger.reload4j

import cn.labzen.logger.kernel.LabzenLogger
import org.slf4j.ILoggerFactory
import org.slf4j.Logger

@Suppress("unused")
class LabzenReload4jLoggerFactory(private val principal: ILoggerFactory) : ILoggerFactory {

  override fun getLogger(name: String?): Logger {
    val original: Logger = principal.getLogger(name)
    return LabzenLogger(original)
  }
}
